src/librustc/middle/infer/region_inference/mod.rs

```diff
@@ -460,6 +468,10 @@ impl<'a, 'tcx> RegionVarBindings<'a, 'tcx> {
             // equating regions.
             self.make_subregion(origin.clone(), sub, sup);
             self.make_subregion(origin, sup, sub);
+
+            if let (ty::ReVar(sub), ty::ReVar(sup)) = (sub, sup) {
+                self.unification_table.borrow_mut().union(sub, sup);
+            }
```

Comment:
Hmm. I expected to not add two subregion edges here, but rather to "refresh" the regions in the inference propagation below.